src/Symfony/Installer/DownloadCommand.php

```diff
@@ -0,0 +1,345 @@
+<?php
+
+/*
+ * This file is part of the Symfony Installer package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Installer;
+
+use Distill\Distill;
+use Distill\Exception\IO\Input\FileCorruptedException;
+use Distill\Exception\IO\Input\FileEmptyException;
+use Distill\Exception\IO\Output\TargetDirectoryNotWritableException;
+use Distill\Strategy\MinimumSize;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Message\Response;
+use GuzzleHttp\Subscriber\Progress\Progress;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\ProgressBar;
+
+/**
+ * Abstract command used by commands which download and extract compressed Symfony files.
+ *
+ * @author Christophe Coevoet <stof@notk.org>
+ * @author Javier Eguiluz <javier.eguiluz@gmail.com>
+ */
+abstract class DownloadCommand extends Command
+{
+    /**
+     * Chooses the best compressed file format to download (ZIP or TGZ) depending upon the
+     * available operating system uncompressing commands and the enabled PHP extensions
+     * and it downloads the file.
+     *
+     * @return Command
+     *
+     * @throws \RuntimeException if the Symfony archive could not be downloaded
+     */
+    protected function download()
+    {
+        $this->output->writeln(sprintf("\n Downloading %s...\n", $this->getDownloadedFileName()));
```

Comment:
same for any other property used here btw (look at Insight which can probably warn you IIRC)